### PR TITLE
feat: User service ports Traefik Docker labels

### PR DESCRIPTION
**Pull request by laurentluce**
## Description:
As part of the authenticated public http ports feature, we are bringing Traefik inside the Docker cluster to route HTTP traffic to the user service HTTP ports.  This PR sets the required Docker labels.

This was tested with Traefik running inside the engine enclave with the following static config and the user service enclave network added to the Traefik container list of networks.

```
version: '3'

services:
  reverse-proxy:
    image: traefik:v2.10
    # Enables the web UI and tells Traefik to listen to docker
    command:
      - --accesslog=true
      - --api.debug=true
      - --api.insecure=true
      - --api.dashboard=true
      - --api.disabledashboardad=true
      - --providers.docker
      - --entrypoints.web.address=:8000
      - --providers.docker.network=bridge
      - --providers.docker.exposedByDefault=false
      - --log.level=DEBUG
    ports:
      # The HTTP port
      - "8000:8000"
      # The Web UI (enabled by --api.insecure=true)
      - "8080:8080"
    volumes:
      # So that Traefik can listen to the Docker events
      - /var/run/docker.sock:/var/run/docker.sock
```

User service "nginx" port labels:
Enclave short UUID: 65d2fb6d6732
Service short UUID: 3771c85af16a
HTTP Port number: 80
```
"traefik.enable": "true"
"traefik.http.routers.65d2fb6d6732-3771c85af16a-80.rule": "Host(`80-3771c85af16a-65d2fb6d6732`)"
"traefik.http.routers.65d2fb6d6732-3771c85af16a-80.service": "65d2fb6d6732-3771c85af16a-80"
"traefik.http.services.65d2fb6d6732-3771c85af16a-80.loadbalancer.server.port": "80"
```

```
$ curl -I http://localhost:8000 -H "Host: 80-3771c85af16a-65d2fb6d6732"
HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 615
Content-Type: text/html
Date: Wed, 29 Nov 2023 21:32:51 GMT
Etag: "6537cac7-267"
Last-Modified: Tue, 24 Oct 2023 13:46:47 GMT
Server: nginx/1.25.
```

## Is this change user facing?
NO

## References (if applicable):
https://www.notion.so/kurtosistech/Public-user-service-HTTP-ports-bdf1107b0d1c4ca990c346fd87473174

### CODE REVIEW

File: container-engine-lib/lib/backend_impls/docker/object_attributes_provider/enclave_object_attributes_provider.go [hunk 1]
```diff
@@ -516,6 +528,94 @@ func (provider *dockerEnclaveObjectAttributesProviderImpl) getLabelsForEnclaveOb
...
+			ruleLabelKey, err := docker_label_key.CreateNewDockerTraefikLabelKey(ruleKeySuffix)
+			if err != nil {
+				return nil, stacktrace.Propagate(err, "An error occurred getting the traefik rule label key with suffix '%v'", ruleKeySuffix)
+			}
+			ruleValue := fmt.Sprintf("Host(`%d-%s-%s`)", portSpec.GetNumber(), shortServiceUuid, shortEnclaveUuid)
+			ruleLabelValue, err := docker_label_value.CreateNewDockerLabelValue(ruleValue)
+			if err != nil {
+				return nil, stacktrace.Propagate(err, "An error occurred creating the traefik rule label value with value '%v'", ruleValue)
+			}
+			labels[ruleLabelKey] = ruleLabelValue
```
Comment (adschwartz) [inline container-engine-lib/lib/backend_impls/docker/object_attributes_provider/enclave_object_attributes_provider.go:572]: @laurentluce I know it's hard to unit test this traefik integration but I wonder if we could break each of these out as functions (host rule, service name  and service port number ) and write some unit tests to verify that they work as intended. It helps lock down the functionality and ensure future engineers have something to test against if they need to change.

Comment (adschwartz) [reply container-engine-lib/lib/backend_impls/docker/object_attributes_provider/enclave_object_attributes_provider.go:572]: Alternatively just test the entire function if that's easier?

Comment (laurentluce) [reply container-engine-lib/lib/backend_impls/docker/object_attributes_provider/enclave_object_attributes_provider.go:572]: I added a test validating the user service labels including the traefik labels.  The user service has two TCP ports, only one HTTP.

Comment (laurentluce) [reply container-engine-lib/lib/backend_impls/docker/object_attributes_provider/enclave_object_attributes_provider.go:572]: I am also planning to add integration tests once Traefik is included.

**Review by adschwartz — APPROVED** [review]

**Pull request merged**